### PR TITLE
ME0 migration to DD4HEP

### PR DESCRIPTION
**Pull request by slomeo**
#### PR description:

This PR is the same of the PR #29609 that was fully signed (by @cvuosalo , @perrotta , @kpedro88 ) but following the request of @silviodonato , in order to solve some conflict originated from #29632 (merged), I have decided to re-start with a new PR within a new IB. The PR #29609 will be closed asap. 

This PR is related to the ME0 migration to DD4HEP
Only code Geometry migration, no alignment, no sensitive detectors

#### PR validation:

1) Made by ME0ValidateGeometryDD4hep_cfg.py and ME0ValidateGeomemtryOldDD_cfg.py.
see histos in #29632 

2) runTheMatrix test ok with wf 18.0

### CODE REVIEW

**Comment by slomeo** [comment]
I closed #29609

File: Geometry/GEMGeometryBuilder/plugins/BuildFile.xml [hunk 1]
```diff
@@ -6,29 +6,37 @@
...
+
 <library name="GEMGeometryPlugins" file="GEMGeometryESModule.cc">
-  <use name="DetectorDescription/Core"/>
-  <use name="DetectorDescription/DDCMS"/>
-  <use name="Geometry/GEMGeometry"/>
-  <use name="dd4hep"/>
-  <lib name="Geom"/>
-  <use name="Geometry/GEMGeometryBuilder"/>
-  <flags EDM_PLUGIN="1"/>
+<use name="DetectorDescription/Core"/>
```
Comment (kpedro88) [inline Geometry/GEMGeometryBuilder/plugins/BuildFile.xml:11]: it looks like you undid the buildfile changes from the recent xml formatting pass, please revert

Comment (slomeo) [reply Geometry/GEMGeometryBuilder/plugins/BuildFile.xml]: @kpedro88 : now "diff new_BuildFile.xml old_BuildFile.xml" is related only to the ME0 Validation part:
< 
< <library name="ME0GeometryValidationPlugins" file="ME0GeometryValidate.cc">
<   <use name="Fireworks/Core"/>
<   <use name="Geometry/GEMGeometry"/>
<   <use name="rootgeom"/>
<   <flags EDM_PLUGIN="1"/>
< </library>

Comment (slomeo) [reply Geometry/GEMGeometryBuilder/plugins/BuildFile.xml]: @kpedro88 : I restored the old format and I added only the ME0 validation part

**Comment by perrotta** [comment]
+1
- Already signed as #29609
- Geometry related pull request: changes for reco are just technical, and involve the replacement of "ME0GeometryBuilderFromDDD" with "ME0GeometryBuilder" in a test script in RecoLocalMuon/GEMRecHit
- Jenkins tests pass

**Comment by kpedro88** [comment]
+upgrade

**Comment by cvuosalo** [comment]
+1

**Comment by silviodonato** [comment]
+operations

**Comment by silviodonato** [comment]
+1

**Pull request merged**